src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -71,11 +72,12 @@ public partial class FunctionInvokingChatClient : DelegatingChatClient
     /// <param name="innerClient">The underlying <see cref="IChatClient"/>, or the next instance in a chain of clients.</param>
     /// <param name="loggerFactory">An <see cref="ILoggerFactory"/> to use for logging information about function invocation.</param>
     /// <param name="functionInvocationServices">An optional <see cref="IServiceProvider"/> to use for resolving services required by the <see cref="AIFunction"/> instances being invoked.</param>
-    public FunctionInvokingChatClient(IChatClient innerClient, ILoggerFactory? loggerFactory = null, IServiceProvider? functionInvocationServices = null)
+    /// <param name="features">An <see cref="IFeatureCollection"/> for acquiring the <see cref="ActivitySource"/> possibly set by earlier <see cref="OpenTelemetryChatClient"/> middleware.</param>
```

Comment:
I'll repeat my feedback previously: this feels like a lot of effort and complication just for the ActivitySource published by OpenTelemetryChatClient. We only did it this way because it happened to work for the needed cases. I'm not seeing the value in generalizing this capability. Where else is it going to be used?

Reply:
I suspect that there will be more cases where earlier "client factories" or "middleware" want to expose state to factories added later in the pipeline. I think it's really odd to make any kind of wrapping pipeline without the ability to pass context from the outer wrappers to the inner wrappers during pipeline construction whether or not you call it middleware.

I think it'd be worth it if all this ever did was make it so you could all UseOpenTelemetry at the start of the ChatClientBuilder pipeline and expose its ActivitySource to all FunctionInvokingChatClient added later in the same pipeline. I also think this simplifies the pipeline by following the typical `List<Func<T, T>> where T : Delegate` pattern.

But if you don't think it's worth it, I'll close the PR.

Reply:
@SteveSandersonMS ?

Reply:
Looking at the implementation, it seems this is something we could add non-breakingly in the future if we get further use cases.

If:

  * it would only be used for otel
  * and there's some alternative in that case (e.g., people can put `UseOpenTelemetry` manually as the outer middleware if they want to measure spans both inside and outside function calling)
  * and unless we also figure out how this would work with Aspire integration

...I'm fine with leaving things as-is.

Reply:
I admit I probably had a too server-centric view of things to think that the otel middleware naturally belongs at the start of the pipeline. And as you say, we could add it later.